### PR TITLE
Replace node with nvm

### PR DESCRIPTION
**Pull request by mandrean**
- Add [NVM](https://github.com/creationix/nvm) instead of just node. NVM then installs the latest stable node.

### CODE REVIEW

File: mac [hunk 1]
```diff
@@ -134,6 +135,44 @@ brew "postgres", restart_service: true
...
 EOF
 
+fancy_echo "Configuring Node ..."
+nvm install stable
+nvm alias default stable
+# shellcheck disable=SC1090
+. "$(brew --prefix nvm)/nvm.sh"
+append_to_zshrc '#NVM'
+# shellcheck disable=SC2016
+append_to_ashrc 'export NVM_DIR="$HOME/.nvm"' 1
```
Comment (angusfretwell) [inline mac]: I think this should be `append_to_zshrc`

**Comment by mandrean** [comment]
@angusfretwell sorry about that! fixed + squashed commits

**Comment by blainesch** [comment]
Node was already included, but python was not. Should these be separate pull requests?

**Comment by mandrean** [comment]
@blainesch Could you be more specific? This PR is more about installing node via nvm instead of just getting the latest version. Devs often have to switch between different node versions...

**Comment by blainesch** [comment]
I'm suggesting adding `pyenv` should be it's own pull request. Node was previously installed by laptop, so that's an upgrade. Python was not, so that's a new feature.

**Comment by mandrean** [comment]
I see! I can do that, np.

**Comment by mandrean** [comment]
@blainesch Sorry for late response! Forgot about this. OK now?

**Comment by andersonvom** [comment]
We're already using `rbenv` and are about to add `pyenv`, so for consistency, wouldn't it be better to use `nodenv` instead of `nvm`?

**Comment by mandrean** [comment]
@andersonvom I disagree... NVM is pretty much the golden standard these days with ~16 000 stars on GitHub vs nodenv with ~300.

**Comment by croaky** [comment]
We used to have nvm in Laptop and removed it in https://github.com/thoughtbot/laptop/pull/363. Asking around to see if others are using nvm at the moment.

**Comment by ihackalot** [comment]
I do use it

**Comment by AlJohri** [comment]
I use `nodenv` regularly and its very stable, installable via homebrew (unlike nvm), and has feature parity with `pyenv` and `rbenv` which made scripting and usage easier for me.

File: mac [hunk 2]
```diff
@@ -134,6 +134,18 @@ brew "postgres", restart_service: true
 brew "redis", restart_service: true
 EOF
 
+fancy_echo "Configuring Node ..."
+nvm install stable
+nvm alias default stable
+# shellcheck disable=SC1090
+. "$(brew --prefix nvm)/nvm.sh"
```
Comment (dplummer) [inline mac]: This needs to come before `nvm install stable` or you won't be able to find the `nvm` command

**Comment by mandrean** [comment]
@dplummer not true. However, `stable` has been deprecated lately and should be replaced with `node` which it is currently an alias for.

**Comment by derekprior** [comment]
I'm working on a branch to use `asdf` for version management. This will let us use one version manager for a whole host of languages and prevent us from having to install a new one with its own conventions for every language we get interested in.

**Comment by derekprior** [comment]
See #502

**Comment by croaky** [comment]
https://github.com/asdf-vm/asdf-plugins Node listed as supported.

**Pull request closed without merging**